unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/FriendsHUD/Scripts/Tabs/FriendsTabComponentView.cs

```diff
@@ -0,0 +1,359 @@
+    [SerializeField] private FriendListComponents onlineFriendsList;
+    [SerializeField] private FriendListComponents offlineFriendsList;
+    [SerializeField] private FriendListComponents searchResultsFriendList;
+    [SerializeField] private SearchBarComponentView searchBar;
+    [SerializeField] private UserContextMenu contextMenuPanel;
+    [SerializeField] private Model model;
+
+    private readonly Dictionary<string, FriendEntryBase.Model> creationQueue =
+        new Dictionary<string, FriendEntryBase.Model>();
+
+    private readonly Dictionary<string, PoolableObject> pooleableEntries = new Dictionary<string, PoolableObject>();
+    private readonly Dictionary<string, FriendEntry> entries = new Dictionary<string, FriendEntry>();
+    private readonly Dictionary<string, ulong> timestamps = new Dictionary<string, ulong>();
+    private Pool entryPool;
+    private string lastSearch;
+
+    public Dictionary<string, FriendEntry> Entries => entries;
+
+    public bool DidDeferredCreationCompleted => creationQueue.Count == 0;
+
+    public event Action<FriendEntry> OnWhisper;
+    public event Action<string> OnDeleteConfirmation;
+
+    public override void OnEnable()
+    {
+        base.OnEnable();
+
+        searchBar.Configure(new SearchBarComponentModel {placeHolderText = "Search friend"});
+        searchBar.OnSearchText += Filter;
+        contextMenuPanel.OnBlock += HandleFriendBlockRequest;
+        contextMenuPanel.OnUnfriend += HandleUnfriendRequest;
+
+        int SortByTimestamp(FriendEntryBase.Model u1, FriendEntryBase.Model u2)
+        {
+            var t1 = timestamps.ContainsKey(u1.userId) ? timestamps[u1.userId] : 0;
+            var t2 = timestamps.ContainsKey(u2.userId) ? timestamps[u2.userId] : 0;
+            return t2.CompareTo(t1);
+        }
+
+        int SortByAlphabeticalOrder(FriendEntryBase.Model u1, FriendEntryBase.Model u2)
+        {
+            return string.Compare(u2.userName, u1.userName, StringComparison.Ordinal);
+        }
+
+        onlineFriendsList.list.SortingMethod = SortByTimestamp;
+        offlineFriendsList.list.SortingMethod = SortByTimestamp;
+        searchResultsFriendList.list.SortingMethod = SortByAlphabeticalOrder;
+    }
+
+    public override void OnDisable()
+    {
+        base.OnDisable();
+
+        searchBar.OnSearchText -= Filter;
+        contextMenuPanel.OnBlock -= HandleFriendBlockRequest;
+        contextMenuPanel.OnUnfriend -= HandleUnfriendRequest;
+    }
+
+    public void Expand()
+    {
+        onlineFriendsList.list.Expand();
+        offlineFriendsList.list.Expand();
+    }
+
+    public void Show()
+    {
+        gameObject.SetActive(true);
+        enabledHeader.SetActive(true);
+        disabledHeader.SetActive(false);
+    }
+
+    public void Hide()
+    {
+        gameObject.SetActive(false);
+        enabledHeader.SetActive(false);
+        disabledHeader.SetActive(true);
+        contextMenuPanel.Hide();
+    }
```

Comment:
You should override the Show/Hide functions already implemented in the `BaseComponentView` class.